### PR TITLE
Add peagle docs

### PR DESCRIPTION
**Pull request by shanjiaz**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Added p-eagle tutorial and algorithm intro

## Tests

I have filled in:

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan/results, such as providing test command and pasting the results.
- [x] (Optional) The necessary documentation update.
- [x] I (a human) have written or reviewed the code in this pr to the best of my ability.

### CODE REVIEW

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# Train P-EAGLE Model Offline
+
+This tutorial walks you through training a P-EAGLE (Parallel EAGLE) speculator model using **offline training**, where hidden states are pre-generated and cached before training begins. This example uses `Qwen/Qwen3-8B` as the target model, but the process is the same for other models. This tutorial follows the same structure as [Train Eagle-3 Offline](train_eagle3_offline.md). The key differences are:
+
+- **Step 2 — vLLM launch:** P-EAGLE requires passing `--hidden-states-path` to `launch_vllm.py` so the server knows where to write hidden state files.
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: This isn't a difference with P-EAGLE

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+- **Step 5 — Training:** P-EAGLE introduces several additional parameters: `--speculator-type peagle`, `--num-depths`, `--down-sample-ratio`, `--down-sample-ratio-min`, `--no-norm-before-residual`, and `--scheduler-type cosine`.
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: `--no-norm-before-residual` isn't specific to P-EAGLE either, is it?

Same with `--scheduler-type`

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+**Time:** ~30 seconds for 5K samples
+
+**Note:** This step is used to setup the dataset that will be used to train your model and is the same for both online and offline training. For more information please see the [prepare_data.py cli reference](/cli/prepare_data.md).
+
+## Step 2: Launch vLLM Server
+
+During training, the drafter model takes internal hidden states from the verifier model as input. We use vLLM to serve the verifier and extract these hidden states. The `launch_vllm.py` script is a lightweight wrapper that sets up the right CLI arguments for vLLM to enable hidden state extraction.
+
+For P-EAGLE, you must pass `--hidden-states-path` to tell the server where to write hidden state files:
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: Not specific

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+
+**Use more GPUs:**
+
+```bash
+# 8 GPUs with DP=8
+python scripts/launch_vllm.py model -- --data-parallel-size 8
+```
+
+**Use multiple nodes:**
+
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: Let's skip the multi-node mention here. I don't think it's needed

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+**Key P-EAGLE-specific parameters:**
+
+- `--speculator-type peagle` - Use the P-EAGLE algorithm
+- `--num-layers 4` - Number of draft transformer layers
+- `--num-depths 4` - Number of parallel prediction depths (multi-token prediction)
+- `--down-sample-ratio 0.7` - Initial down-sampling ratio for COD sampling
+- `--down-sample-ratio-min 0.2` - Minimum down-sampling ratio for COD sampling
+- `--no-norm-before-residual` - Disable normalization before residual connections
+- `--scheduler-type cosine` - Use cosine learning rate schedule
+- `--on-missing raise` - Fail if any hidden states are missing (recommended). Alternatives are `skip` and `warn` which both skip the missing sample, with the latter raising a warning.
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: These aren't specific

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+## Step 7: Test Your Model
+
+### Quick Test with vLLM
+
+Stop the training vLLM server (Ctrl+C), then serve your speculator:
+
+```bash
+# in vllm venv
+vllm serve ./checkpoints/checkpoint_best --port 8000
+```
```
[the remark this answers is not in the document]
Comment (fynnsu) [reply docs/user_guide/tutorials/train_peagle_offline.md:267]: Yeah, we should fix this

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+
+Per-position acceptance:
+
+| Position | Acceptance |
+| -------- | ---------- |
+| 0        | 40.84%     |
+| 1        | 10.84%     |
+| 2        | 1.58%      |
+| 3        | 0.15%      |
+
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md:299]: Are these numbers real?

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+   --epochs 20  # Train longer
+   ```
+
+## Next Steps
+
+After training your model:
+
+1. **Evaluate performance** - See [Evaluating Performance](evaluating_performance.md)
+2. **Deploy to production** - See [Serve in vLLM](serve_vllm.md)
+3. **Fine-tune further** - Use `--from-pretrained ./checkpoints/latest` to continue training
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: Wrong path here as well.

**Review by fynnsu — CHANGES_REQUESTED** [review]
Some comments on tutorial. Otherwise looks good

File: docs/user_guide/tutorials/train_peagle_offline.md (hunk 2 above)
Comment (shanjiaz) [reply docs/user_guide/tutorials/train_peagle_offline.md]: yeah

File: docs/user_guide/tutorials/train_peagle_offline.md (hunk 4 above)
Comment (shanjiaz) [reply docs/user_guide/tutorials/train_peagle_offline.md]: sure!

File: docs/user_guide/tutorials/train_peagle_offline.md (hunk 7 above)
Comment (shanjiaz) [reply docs/user_guide/tutorials/train_peagle_offline.md:299]: yeah drawn from the example.

File: docs/user_guide/tutorials/train_peagle_offline.md (hunk 8 above)
Comment (shanjiaz) [reply docs/user_guide/tutorials/train_peagle_offline.md]: updated

File: docs/user_guide/tutorials/train_peagle_offline.md [hunk 9]
```diff
@@ -250,13 +229,11 @@ CUDA_VISIBLE_DEVICES=0,1 torchrun \
 **Key P-EAGLE-specific parameters:**
 
 - `--speculator-type peagle` - Use the P-EAGLE algorithm
-- `--num-layers 4` - Number of draft transformer layers
 - `--num-depths 4` - Number of parallel prediction depths (multi-token prediction)
 - `--down-sample-ratio 0.7` - Initial down-sampling ratio for COD sampling
 - `--down-sample-ratio-min 0.2` - Minimum down-sampling ratio for COD sampling
 - `--no-norm-before-residual` - Disable normalization before residual connections
 - `--scheduler-type cosine` - Use cosine learning rate schedule
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train_peagle_offline.md]: Remove this one (scheduler-type) too

**Review by fynnsu — APPROVED** [review]
Looks good! You still have one unrelated cli arg, but then it can be merged.

File: docs/user_guide/tutorials/train_peagle_offline.md (hunk 9 above)
Comment (shanjiaz) [reply docs/user_guide/tutorials/train_peagle_offline.md]: sure!
